doc/devel/style_guide.rst

```diff
@@ -0,0 +1,268 @@
+| The fundamental ideas behind       | Matplotlib will take data and      |
+| Matplotlib for visualization       | transform it through functions and |
+| involve taking data and            | methods. They can generate many    |
+| transforming it through functions  | kinds of visuals. These will be the|
+| and methods.                       | fundamentals for using Matplotlib. |
++------------------------------------+------------------------------------+
+
+Voice
+^^^^^
+Write in active sentences. Passive voice is best for situations or conditions
+related to warning prompts.
+
++------------------------------------+------------------------------------+
+| Correct                            | Incorrect                          |
++====================================+====================================+
+| The function ``plot`` generates the| The graph is generated by the      |
+| graph.                             | ``plot`` function.                 |
++------------------------------------+------------------------------------+
+| An error message is returned by the| You will see an error message from |
+| function if there are no arguments.| the function if there are no       |
+|                                    | arguments.                         |
++------------------------------------+------------------------------------+
+
+Sentence structure
+^^^^^^^^^^^^^^^^^^
+Write with short sentences using Subject-Verb-Object order regularly. Limit
+coordinating conjunctions in sentences. Avoid pronoun references and
+subordinating conjunctive phrases.
+
++------------------------------------+------------------------------------+
+| Correct                            | Incorrect                          |
++====================================+====================================+
+| The ``pyplot`` module in Matplotlib| The ``pyplot`` module in Matplotlib|
+| is a collection of functions. These| is a collection of functions which |
+| functions create, manage, and      | create, manage, and manipulate the |
+| manipulate the current Figure and  | current Figure and plotting area.  |
+| plotting area.                     |                                    |
++------------------------------------+------------------------------------+
+| The ``plot`` function plots data   | The ``plot`` function plots data   |
+| to the respective Axes. The Axes   | within its respective Axes for its |
+| corresponds to the respective      | respective Figure.                 |
+| Figure.                            |                                    |
++------------------------------------+------------------------------------+
+| The implicit approach is a         | Users that wish to have convenient |
+| convenient shortcut for            | shortcuts for generating plots use |
+| generating simple plots.           | the implicit approach.             |
++------------------------------------+------------------------------------+
+
+
+Formatting
```

Comment:
This section is similar to https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages. Maybe they can be merged.

Reply:
I do reference this page with the `See also` link at the top. I think it makes sense later down the line for us to incorporate the relevant sections.